mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -191,6 +191,8 @@ public Tuple getTuple(EntityKey key, TupleContext tupleContext) {
 			for ( int i = 0; i < searchObjects.length; i++ ) {
 				if ( dbObject.get( ID_FIELDNAME ).equals( searchObjects[i] ) ) {
 					tuples.set( i, createTuple( keys[i], tupleContext, dbObject ) );
```

Comment:
> that's not premature optimisation, that's just pointless overhead.

It's needed for this approach. Out of curiosity, would it be better to use an array instead of list in this case? Something like:
```
Tuple[] tuples = new Tuple[ keys.length];
...
return Arrays.asList( tuples );
```